valhalla/meili/viterbi_search.h

```diff
@@ -155,27 +163,47 @@ class IViterbiSearch
   stateid_iterator PathEnd() const
   { return path_end_; }
 
+  const IEmissionCostingMode& emission_costing_mode() const
+  { return emission_costing_mode_; }
+
+  void set_emission_costing_mode(const IEmissionCostingMode costing_mode)
+  { emission_costing_mode_ = costing_mode; }
+
+  const ITransitionCostingMode& transition_costing_mode() const
+  { return transition_costing_mode_; }
+
+  void set_transition_costing_mode(const ITransitionCostingMode costing_mode)
+  { transition_costing_mode_ = costing_mode; }
+
   virtual StateId Predecessor(const StateId& stateid) const = 0;
 
   virtual double AccumulatedCost(const StateId& stateid) const = 0;
 
  protected:
   // Calculate transition cost from left state to right state
-  virtual float TransitionCost(const StateId& lhs, const StateId& rhs) const = 0;
+  virtual float TransitionCost(const StateId& lhs, const StateId& rhs) const
+  { return transition_costing_mode_(lhs, rhs); }
```

Comment:
These virtual methods are still here but default to invoke the corresponding costing mode.